pkg/storage/batch.go

```diff
@@ -510,7 +567,7 @@ func fetchChunkBySeries(ctx context.Context, chunks []*LazyChunk, matchers []*la
 
 	// Now that we have the first chunk for each series loaded,
 	// we can proceed to filter the series that don't match.
-	chksBySeries = filterSeriesByMatchers(chksBySeries, matchers)
+	chksBySeries = filterSeriesByMatchers(chksBySeries, matchers, metrics)
 
 	var allChunks []*LazyChunk
 	for _, series := range chksBySeries {
```

Comment:
Wondering what a good way would be to also track how many series there were.

I'm not sure this is a fully vetted thought yet but part of me thinks we will want to know something about the number of series involved too.